File: ios/OpenClawConsole/OpenClawConsole/Models/ChatMessage.swift
// Models/ChatMessage.swift
// OpenClaw Work Console
// ChatMessage model matching protocol.md

import Foundation

// MARK: - Message Role

enum MessageRole: String, Codable {
    case user
    case agent

    var isUser: Bool { self == .user }
}

// MARK: - ChatMessage

struct ChatMessage: Codable, Identifiable, Hashable {
    let id: String
    let agentId: String
    let taskId: String?
    let role: MessageRole
    let content: String
    let timestamp: Date

    enum CodingKeys: String, CodingKey {
        case id
        case agentId = "agent_id"
        case taskId = "task_id"
        case role
        case content
        case timestamp
    }

    static func == (lhs: ChatMessage, rhs: ChatMessage) -> Bool {
        lhs.id == rhs.id
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
}

// MARK: - Outbound Chat Request

struct ChatMessageRequest: Codable {
    let agentId: String
    let message: String
    let taskId: String?

    enum CodingKeys: String, CodingKey {
        case agentId = "agent_id"
        case message
        case taskId = "task_id"
    }
}

File: openclaw-skills/android/app/src/main/java/com/openclaw/console/data/model/ChatMessage.kt
package com.openclaw.console.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class MessageRole {
    @SerialName("user") USER,
    @SerialName("agent") AGENT
}

@Serializable
data class ChatMessage(
    val id: String,
    @SerialName("agent_id") val agentId: String,
    @SerialName("task_id") val taskId: String? = null,
    val role: MessageRole,
    val content: String,
    val timestamp: String
)
